nntrainer/layers/layer_internal.h

```diff
@@ -126,6 +126,19 @@ class Layer {
    */
   virtual void calcDerivative() = 0;
 
+  /**
+   * @brief  check if this layer requires label to be passed
+   * @note   if requireLabel() == true means, for now, that it is endpoint of a
+   * graph(numOutlayers == 0). label will be fed to the gradient of hidden if
+   * requireLabel is true
+   * @todo   If we get to have a use case for requireLabel(true) but in the
+   * middle of a graph, change the semantics
+   *
+   * @return true requires a label when training
+   * @return false does not require a label
```

Comment:
If you describe w/ a return value and its description, @retval is more appropriate. 

In general, use @return if you describe without any structure, use @retval if you pair up "return value" with its description.

Reply:
Thank you for the explanation :). There are number of this issue in other places, so I'll make a separate PR for this.

Reply:
#1264 deals the issue :)